### ISSUE TITLE
authenticated redirects should disable revival

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1078]
see `ikev2-redirect-03-auth-loop` where the connection ends up both trying to revive and trying to redirect:
```
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: initiating IKEv2 connection
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: sent IKE_SA_INIT request to 192.1.2.23:500
1v2 "westnet-eastnet-ipv4-psk-ikev2" #1: sent IKE_AUTH request {cipher=AES_GCM_16_256 integ=n/a prf=HMAC_SHA2_512 group=MODP2048}
003 "westnet-eastnet-ipv4-psk-ikev2" #1: initiator established IKE SA; authenticated peer using authby=secret and ID_FQDN '@east'
002 "westnet-eastnet-ipv4-psk-ikev2" #1: initiating IKE_AUTH redirect to new gateway (address: 192.1.3.33)
1v2 "westnet-eastnet-ipv4-psk-ikev2" #3: initiating IKEv2 connection
002 "westnet-eastnet-ipv4-psk-ikev2" #1: IKE SA expired (LATEST!)
002 "westnet-eastnet-ipv4-psk-ikev2" #1: deleting IKE SA (ESTABLISHED_IKE_SA) and sending notification
002 "westnet-eastnet-ipv4-psk-ikev2" #1: connection is supposed to remain up; revival attempt 1 scheduled in 0 seconds
1v2 "westnet-eastnet-ipv4-psk-ikev2" #3: sent IKE_SA_INIT request to 192.1.3.33:500
1v2 "westnet-eastnet-ipv4-psk-ikev2" #3: sent IKE_AUTH request {cipher=AES_GCM_16_256 integ=n/a prf=HMAC_SHA2_512 group=MODP2048}
003 "westnet-eastnet-ipv4-psk-ikev2" #3: initiator established IKE SA; authenticated peer using authby=secret and ID_FQDN '@east'
002 "westnet-eastnet-ipv4-psk-ikev2" #3: initiating IKE_AUTH redirect to new gateway (address: 192.1.2.23)
1v2 "westnet-eastnet-ipv4-psk-ikev2" #5: initiating IKEv2 connection
002 "westnet-eastnet-ipv4-psk-ikev2" #3: IKE SA expired (LATEST!)
002 "westnet-eastnet-ipv4-psk-ikev2" #3: deleting IKE SA (ESTABLISHED_IKE_SA) and sending notification
002 "westnet-eastnet-ipv4-psk-ikev2" #3: connection is supposed to remain up; revival attempt 2 scheduled in 5 seconds
```

**Comment by vukasink** [comment]
Hmmm, I am trying to understand, why only in authenticated redirects? In `ikev2-redirect-01-global` I see also:

``002 "westnet-eastnet-ipv4-psk-ikev2" #1: connection is supposed to remain up; revival attempt 1 scheduled in 0 seconds``

Shouldn't it be disabled there as well?

**Comment by cagney** [comment]
> Hmmm, I am trying to understand, why only in authenticated redirects?

I'd not noticed global redirects doing the same thing.

One thing notable about authenticated redirects is that they happen after both ends have proven their identity (i.e., the redirect can be trusted).  Is that true of global redirects?

**Comment by letoams** [comment]
On Sun, 7 May 2023, cagney wrote:

>       Hmmm, I am trying to understand, why only in authenticated redirects?
> 
> I'd not noticed global redirects doing the same thing.
> 
> One thing notable about authenticated redirects is that they happen after both ends have proven their identity (i.e., the redirect
> can be trusted). Is that true of global redirects?

No, global redirects are not trusted.

**Comment by cagney** [comment]
Also fixed by b785389711ec59b83c6e257147fa2a92be95e761

**Comment by cagney** [comment]
> > One thing notable about authenticated redirects is that they happen after both ends have proven their identity (i.e., the redirect can be trusted). Is that true of global redirects?

> No, global redirects are not trusted.

right, the global case should go something like

- initiate to "busy", get redirected to "bored"
- initiate to "bored", fail
- schedule revival starting with "busy"

it was instead scheduling an immediate revival to "busy" simultaneous to initiating to "bored"
